### ISSUE TITLE
Game Studio spiking to 100% CPU usage every few minutes.

### ISSUE THREAD

**Issue by Quahu** [stride3d/stride#913]
**Release Type**: Official Release

**Version**: 4.0.0.1-beta04-1317

**Platform(s)**: Windows

**Describe the bug**
Every few minutes of using the Game Studio the process starts to use 100% of the CPU for ~5-10 seconds, locking everything up.
The spikes don't seem to be very consistent, but I was able to get at least one every 5-15 minutes.

**To Reproduce**
1. Open any project in the Game Studio.
2. Wait a couple of minutes to hopefully see them occur.

**Expected behavior**
-

**Screenshots**
-

**Log and callstacks**
-

**Additional context**
PC Specs:
OS Name: Microsoft Windows 10 Pro
Version: 10.0.18363 Build 18363
Architecture: x64
Processor: Intel(R) Core(TM) i5-4690K CPU @ 3.50GHz, 3501 Mhz, 4 Core(s), 4 Logical Processor(s)
Memory: 16 GB DDR3 1600Mhz

**Comment by manio143** [comment]
I'm experiencing the same issue on master build. Every few minutes (sometimes every minute) the whole system freezes. I don't believe this is happening when GameStudio is the only thing running on my PC (or with just project's VS), but rather when I'm at 70% used RAM or higher.
OS: Window 10 x64 2004 (19041)
Processor: i5-6300U @ 2.40GHz (laptop)
Memory: 8GB

**Comment by Eideren** [comment]
Running gamestudio on core or framework ?

**Comment by Quahu** [comment]
The issue occurs on both.

**Comment by daredloco** [comment]
I'm experiencing this issue as well using NET Framework

OS: Windows 8.1 x64 v6.3 (Build 9600)
Processor: i7-5500U @ 2.40GHz
Memory: 12gb

**Comment by Eideren** [comment]
I was able to repro it, it's fairly consistent when at or under 4 threads. I'll take care of this

**Comment by Eideren** [comment]
Fetch and build latest, I pushed a small workaround which should prevent the freeze to happen for now while I investigate further

**Comment by Shpendicus** [comment]
For my version : 4.0.0.1-beta04-1347 I cant repro this and i have worked with stride  for hours and played around with 3D examples and 5 more scripts soooo can it be considered fix I guess?

**Comment by Quahu** [comment]
> For my version : 4.0.0.1-beta04-1347 I cant repro this and i have worked with stride for hours and played around with 3D examples and 5 more scripts soooo can it be considered fix I guess?

No, as the commit (b7ed1ef) says it's just a temporary solution for the issue which I presume is still unknown.

**Comment by Eideren** [comment]
Yes, it's kind of a pain in the ass to figure out as hooking any profiling tool just make the issue disappear so I have to manually check timings and such which takes ages

**Comment by Shpendicus** [comment]
Actually, I also feel sadly that Stride is taking abit to much of memory is this intended by design (~1GB) RAM? unity3D takes around 600 up to1GB at kind of big project, would be nice to toggle this too somehow abuit down.

**Comment by Eideren** [comment]
@Shpendicus Please, that's unrelated to the issue at hand, move it to another issue instead if you want to discuss this

**Comment by Shpendicus** [comment]
@Eideren sry ^^ Will do so.
